addon-test-support/@ember/test-helpers/dom/trigger-event.js

```diff
@@ -4,14 +4,39 @@ import settled from '../settled';
 import { nextTickPromise } from '../-utils';
 
 /**
-  Triggers an event on the specified target.
-
-  @public
-  @param {string|Element} target the element or selector to trigger the event on
-  @param {string} eventType the type of event to trigger
-  @param {Object} options additional properties to be set on the event
-  @return {Promise<void>} resolves when the application is settled
-*/
+ * Triggers an event on the specified target.
+ *
+ * For HTML `input` element of type 'file', if you wish to trigger `change`
+ * event, the options param can be passed as:
+ *   1. an array of `File` objects.
+ *   2. an object with key `files` which contains an array of `File` objects.
+ *
+ * @public
+ * @param {string|Element} target the element or selector to trigger the event on
+ * @param {string} eventType the type of event to trigger
+ * @param {Object} options additional properties to be set on the event
+ * @return {Promise<void>} resolves when the application is settled
+ *
+ * @example <caption>Usage</caption>
+ *
+ * import { setupRenderingTest } from 'ember-qunit';
+ * import { render, triggerEvent } from '@ember/test-helpers';
+ *
+ * module('awesome-sauce', function(hooks) {
+ *   setupRenderingTest(hooks);
+ *     test('does something awesome', async function(assert) {
+ *     await render(hbs`{{awesome-sauce}}`);
+ *
+ *     // passing an array of files
+ *     let file = new File(['text file'], 'text.txt', { type: 'text/plain' });
+ *     await triggerEvent('input:file', 'change', [file]);
```

Comment:
I think ultimately, we should only have one format here (probably the `{ files: [] }` one?)...